### ISSUE TITLE
IF node mixes up item references for following nodes

### ISSUE THREAD

**Issue by nepomuc** [n8n-io/n8n#7487]
When you use an IF node to sort multiple items and bring in order to process them differently, they loose their references to the items they were linked to before the IF node. Because this is hard for me to explain I built the following flow to reproduce the issue.

**To Reproduce**
Look at and run the following flow:  [IF_node_mixes_up_item_references_for_following_nodes.json](https://github.com/n8n-io/n8n/files/13061059/IF_node_mixes_up_item_references_for_following_nodes.json)

**Expected behavior**
The refences should be consistent as expected.

**Comment by Joffcom** [comment]
Hey @nepomuc,

You are using the older syntax for referencing the data, If you use `$('Three items').item.json` instead it will work. You can find the documentation on this here: https://docs.n8n.io/code/builtin/output-other-nodes/

To use the older style were you updating from an old n8n version or using an old template?

**Comment by nepomuc** [comment]
Hey @Joffcom,

Thanks a lot for the quick response. It's really embarrassing but thanks to your answer I got curious why my n8n still uses the old syntax and I realized that I am bunch of releases behind 😬 My upgrade script seems to have stopped working and since it's just a local playground setup I didn't really check if everything was working flawlessly. I will work though https://github.com/n8n-io/n8n/blob/master/packages/cli/BREAKING-CHANGES.md and upgrade my system step by step now.
Thanks again and sorry for the hacky upgrade process that produced this ticket.
